src/bootloader/bootloaderlite.cc

```diff
@@ -93,21 +95,14 @@ BootloaderLite::VersionStrRes BootloaderLite::getDeploymentVersion(const std::st
 std::string BootloaderLite::getVersion(const std::string& deployment_dir, const std::string& hash,
                                        const std::string& ver_file) {
   try {
-    std::string file;
-    for (auto& p : boost::filesystem::directory_iterator(deployment_dir)) {
-      std::string dir = p.path().string();
-      if (!boost::filesystem::is_directory(dir)) {
-        continue;
-      }
-      if (boost::algorithm::contains(dir, hash)) {
-        file = dir + ver_file;
-        break;
-      }
-    }
-    if (file.empty()) {
-      LOG_WARNING << "Target deployment hash was not found";
+    std::string target_dir = getTargetDir(deployment_dir, hash);
+
+    if (target_dir.empty() || !boost::filesystem::is_directory(target_dir)) {
```

Comment:
I think `getTargetDir` guarantees that the return path is a dir if it is not empty. So, I would just check if the `target_dir` is empty and print the Warning message saying that the target deployment dir is not found for `hash` in `deployment_dir` (deployments top level root dir).

Reply:
fixed, thanks